src/common/iconv.c

```diff
@@ -67,18 +67,27 @@ int guac_iconv(guac_iconv_read* reader, const char** input, int in_remaining,
     while (in_remaining > 0 && out_remaining > 0) {
 
         int value;
+        int count;
         const char* read_start;
         char* write_start;
 
         /* Read character */
         read_start = *input;
         value = reader(input, in_remaining);
-        in_remaining -= *input - read_start;
+        count = *input - read_start;
+        /* Stop if not enough bytes to read */
```

Comment:
This test is checking whether the reader successfully read a character, not whether enough bytes were read.